interactor/src/mvx_esdt_safe/mvx_esdt_safe_interactor_main.rs

```diff
@@ -21,60 +34,276 @@ impl MvxEsdtSafeInteract {
             .await
             .use_chain_simulator(config.use_chain_simulator());
 
-        interactor.set_current_dir_from_workspace("mvx-esdt-safe");
-        let wallet_address = interactor.register_wallet(test_wallets::alice()).await;
+        interactor.set_current_dir_from_workspace("interactor");
+        let wallet_address = interactor.register_wallet(test_wallets::mike()).await;
+        let bob_address = interactor.register_wallet(test_wallets::bob()).await;
 
         // Useful in the chain simulator setting
         // generate blocks until ESDTSystemSCAddress is enabled
-        interactor.generate_blocks_until_epoch(1).await.unwrap();
+        interactor.generate_blocks_until_epoch(1u64).await.unwrap();
 
-        let contract_code = BytesValue::interpret_from(
-            "mxsc:../output/mvx-esdt-safe.mxsc.json",
-            &InterpreterContext::default(),
-        );
+        let set_state_response = interactor.set_state_for_saved_accounts().await;
+        interactor.generate_blocks(2u64).await.unwrap();
+        assert!(set_state_response.is_ok());
 
         MvxEsdtSafeInteract {
             interactor,
             wallet_address,
-            contract_code,
+            bob_address,
             state: State::load_state(),
         }
     }
 
-    pub async fn deploy(
+    pub fn assert_expected_log(&mut self, logs: Vec<Log>, expected_log: &str) {
+        let expected_bytes = ManagedBuffer::<StaticApi>::from(expected_log).to_vec();
+
+        let found_log = logs.iter().find(|log| {
+            log.topics.iter().any(|topic| {
+                if let Ok(decoded_topic) = BASE64.decode(topic) {
+                    decoded_topic == expected_bytes
+                } else {
+                    false
+                }
+            })
+        });
+
+        assert!(found_log.is_some(), "Expected log not found");
+    }
+
+    pub fn assert_expected_error_message(
+        &mut self,
+        response: Result<(), TxResponseStatus>,
+        expected_error_message: Option<&str>,
+    ) {
+        match response {
+            Ok(_) => assert!(
+                expected_error_message.is_none(),
+                "Transaction was successful, but expected error"
+            ),
+            Err(error) => {
+                assert_eq!(expected_error_message, Some(error.message.as_str()))
+            }
+        }
+    }
+
+    // Arguments should be in plain text, they will be converted to hex
+    pub async fn check_account_storage(
+        &mut self,
+        address: Address,
+        wanted_key: &str,
+        expected_value: Option<&str>,
+    ) {
+        let pairs = self.interactor.get_account_storage(&address).await;
+
+        let wanted_hex = hex::encode(wanted_key);
+
+        let found_entry = pairs.iter().find(|(key, _)| key.contains(&wanted_hex));
+
+        match (found_entry, expected_value) {
```

Comment:
- `(Some(_), None)` can be a valid case, you can just find a key and not check its value (expected_value is Option)
- `(None, None)` should not be a valid case

I think it's too complicated this `match`. My suggestion:
- assert found_entry is_some, "Expected key containing '{}' was not found in account storage." -> with this you make sure the key is in account because that's why you call the function
- if expected_value is_some => assert found_entry value == expected_value -> with this you make sure the value is correct, if sent in the arguments.